### PR TITLE
[Dash-Sonic] Introduce route group, pa validation tables

### PR DESCRIPTION
**Pull request by prsunny**
1. Add Route group table for grouping routes and binding to an ENI
2. Introduce PA validation table for special validation for certain scenarios

### CODE REVIEW

File: documentation/general/dash-sonic-hld.md [hunk 1]
```diff
@@ -494,8 +495,25 @@ vm_vni                   = VM VNI that is used for setting direction. Also used
 
 ### 3.2.9 ROUTE LPM TABLE - OUTBOUND
 
+```
+DASH_ENI_ROUTE_TABLE:{{eni}}:
```
Comment (r12f) [inline documentation/general/dash-sonic-hld.md:499]: nit:
- remove the last ":"
- update eni to "eni_id"

File: documentation/general/dash-sonic-hld.md [hunk 2]
```diff
@@ -494,8 +495,25 @@ vm_vni                   = VM VNI that is used for setting direction. Also used
...
+
+```
+key                      = DASH_ENI_ROUTE_TABLE:eni ; ENI name as key; ENI can bind to a route group table atomically (Overwrites any old bindings)
+; field                  = value 
+group_id                 = Route Group Table Id (contains both IPv4 and IPv6 routes)
+```
+
+```
+DASH_ROUTE_GROUP_TABLE:{{group_id}}
+    "guid": {{string}}
```
Comment (r12f) [inline documentation/general/dash-sonic-hld.md:511]: do we still need the guid here? @sharmasushant

File: documentation/general/dash-sonic-hld.md [hunk 3]
```diff
@@ -1213,7 +1257,20 @@ For the inbound direction, after Route/ACL lookup, pipeline shall use the "under
         "OP": "SET"
     },
     {
-        "DASH_ROUTE_TABLE:F4939FEFC47E:50.1.2.0/24": {
+        "DASH_ENI_ROUTE_TABLE:F4939FEFC47E": {
+	    "group_id":"group_id_2"
+        },
```
Comment (kperumalbfn) [inline documentation/general/dash-sonic-hld.md:1262]: Different ENI ids for group_id_1, group_id_2 and group_id_3.

**Comment by KrisNey-MSFT** [comment]
Currently the route table is modified w/routes belonging to an ENI (per ENI).
Controller calls each route update w/ENI key.
Group table & binding allows for **atomic** binding for an ENI (instead of updating 1 by 1)
(Not a subset of ENI routes).
Cx update their routes in the route table, and expect all or nothing to work.  Adding 1 by 1 we can end up in a non-complete/inconsistent goal state.  

What if libsai has no resources to support an atomic update?  
The controller will need to wait to determine whether this is a successful operation or not.
If unsuccessful, old route table will persist.

File: documentation/general/dash-sonic-hld.md (hunk 1 above)
Comment (prsunny) [reply documentation/general/dash-sonic-hld.md]: removed `:`, we use `eni `in other places, so IMO, we can stick to the same format

File: documentation/general/dash-sonic-hld.md (hunk 2 above)
Comment (prsunny) [reply documentation/general/dash-sonic-hld.md:511]: guid can be optional

Comment (sharmasushant) [reply documentation/general/dash-sonic-hld.md:511]: I don't see a use for guid field, but I am okay to leave it there.

**Review by sharmasushant — APPROVED** [review]
LGTM

**Comment by KrisNey-MSFT** [comment]
Are all comments resolved?  IF Y, I can merge :)

**Comment by KrisNey-MSFT** [comment]
hi @r12f , @kperumalbfn , and @pranjal-git - would you mind reviewing for Prince please?
TY...

File: documentation/general/dash-sonic-hld.md (hunk 3 above)
Comment (prsunny) [reply documentation/general/dash-sonic-hld.md:1262]: This is a reference example. it can be any group_id

**Review by kperumalbfn — APPROVED** [review]

**Review by r12f — APPROVED** [review]

**Pull request merged**